settings_dialog.js

```diff
@@ -1,41 +1,60 @@
 // Everything under the settings button
 "use strict";
 
+const fs = require('fs-extra');
 const path = require('path');
 
 var {shell} = require('electron');
+const {dialog} = require('electron').remote;
+const win = remote.getCurrentWindow();
+
+const {getVersionData} = require("./version_info.js");
 
 const { Modal, showGenericError} = require('./modal');
-const {listInstalledVersions, deleteInstalledVersion} = require('./install_handler.js');
-const { settings, saveSettings, installPath } = require('./settings.js');
+const { listInstalledVersions, deleteInstalledVersion } = require('./install_handler.js');
+
+const { settings, dataFolder, saveSettings, defaultInstallPath } = require('./settings.js');
 
 const settingsModal = new Modal("settingsModal", "settingsModalDialog", {
     closeButton: "settingsClose"
 });
 
+const movingFileModal = new Modal("movingFileModal", "movingFileModalDialog", {
+    autoClose: false
+});
+
+let installedVersions = [];
```

Comment:
I think that this global variable could be avoided by just calling the version finding code whenever the list is needed (and/or passing it around as a parameter).

Reply:
May I just change it to a local variable?

Reply:
Yes, you can do that.